job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/properties/MavenPropertiesContext/mavenInfo.groovy

```diff
@@ -0,0 +1,19 @@
+mavenJob('example_1') {
+    properties {
+        mavenInfo {
+            assignName('name template')
+        }
+    }
+}
+
+mavenJob('example_2') {
+    properties {
+        mavenInfo {
+            assignDescription {
+                descriptionTemplate('some text')
+                modulePattern(':my-artifact')
```

Comment:
Aren't the module and dependencies patterns independent from the description option? You can click the "Advanced..." button even if "Assign Description" is not selected. So the DSL should be like

```groovy
mavenInfo {
    assignName('name template')
    assignDescription('some text')
    modulePattern(':my-artifact')
    interestingDependenciesPattern('org.springframework.*:*')
}
```

Those settings are also used with "assignName", see https://github.com/jenkinsci/maven-info-plugin/blob/master/src/main/java/jenkins/plugins/maveninfo/extractor/properties/NamePropertiesFinder.java#L21-L22 for an example.